onnxruntime/core/common/logging/capture.cc

```diff
@@ -27,16 +27,31 @@ void Capture::ProcessPrintf(msvc_printf_check const char* format, va_list args)
   char message_buffer[kMaxMessageSize];
   const auto message = gsl::make_span(message_buffer);
 
+  bool error = false;
+  bool truncated = false;
+
 #if (defined(WIN32) || defined(_WIN32) || defined(__WIN32__) && !defined(__GNUC__))
   const int nbrcharacters = vsnprintf_s(message.data(), message.size(), _TRUNCATE, format, args);
+  if (nbrcharacters < 0) {
+    // this can fail and return -1 if
+    //   buffer is nullptr (not possible, local buffer),
+    //   format is nullptr (possible),
+    //   count < 0 and != _TRUNCATE (not possible, always == _TRUNCATE)
+    //   or buffer too small and count != _TRUNCATE (not possible, always == _TRUNCATE)
+    // given there's only one possible cause, check that. alternatively we'd have to check errno which isn't threadsafe
```

Comment:
What do you mean not thread safe?

Reply:
Well it used to be many years ago, but apparently that has changed. :) I'll update to use errno.